src/WrappedMToken.sol

```diff
@@ -595,6 +579,60 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
         return UIntMath.safe240(startingBalance_ - IMTokenLike(mToken).balanceOf(address(this)));
     }
 
+    /**
+     * @dev   Starts earning for `account` if allowed by the Registrar.
+     * @param account_      The account to start earning for.
+     * @param currentIndex_ The current index.
+     */
+    function _startEarningFor(address account_, uint128 currentIndex_) internal {
+        _revertIfNotApprovedEarner(account_);
+
+        Account storage accountInfo_ = _accounts[account_];
+
+        if (accountInfo_.isEarning) return;
+
+        accountInfo_.isEarning = true;
+        accountInfo_.lastIndex = currentIndex_;
+
+        uint240 balance_ = accountInfo_.balance;
+
+        _addTotalEarningSupply(balance_, currentIndex_);
+
+        unchecked {
+            totalNonEarningSupply -= balance_;
+        }
```

Comment:
Could be nice to add a comment explaining why it can't underflow.

Reply:
Perrhaps, but we have 4 of these unchecked decrements in the code, and they are unrelated to this PR (this code was just moved/internalized, not changed).